### ISSUE TITLE
User Management of Nodes - Hide

### ISSUE THREAD

**Issue by happyconcepts** [bitshares/bitshares-ui#1570]
I would like to manage the nodes in my client a bit more carefully.

In order to deal with the [OpenLedger situation about compromised domains and certificates](https://steemit.com/bitshares/@themarkymark/psa-open-ledger-hacked), I would like to prevent the client from trying to connect to specific domains. I anticipate changing my settings back to normal once it is sorted out.

In the Settings>Access page I quickly found two nodes using the openledger.info domain so I chose to "hide" them...and there is also a node with an openledger.hk address that I am unsure about what to do.

What does "hide" mean, exactly?

**Comment by startailcoon** [comment]
Hidden nodes are **not** used for automatic connecting. You can always decide to connect to them manually without hiding them.

**Comment by happyconcepts** [comment]
So I would like to believe you @startailcoon but the software is not behaving as you describe.

Despite being "hidden" by me, at least one openledger node is contacted by the bitshares client.
So my question remains unanswered?

![ol_connects](https://user-images.githubusercontent.com/14145836/41129710-37cac15c-6a68-11e8-8f27-7195db04d063.png)

**Comment by svk31** [comment]
What makes you think that node is being contacted?

**Comment by clockworkgr** [comment]
I think he means that hidden nodes shouldn't even report latency/not be connected to at all.

**Comment by happyconcepts** [comment]
Yes it would seem to me that there is no other way to assess the latency except by making a connection. 

>You can always decide to connect to them manually without hiding them.

@startailcoon Thanks but this is obvious. Do you understand my question? 

You opened #1222 titled *Allow wallet to connect to hidden nodes when none other are available* and yet ... 

Such very very timing with the OL steal.

**Comment by svk31** [comment]
Latency stats are stored in localStorage, so if it's been connected to at some point in the past it will have a latency number. That doesn't mean it's being connected to while hidden...

**Comment by happyconcepts** [comment]
Ok I understand you @svk31 and thanks!  But it just seems messier the more I look.

Here is even further unexpected behavior that just occurred and likely related; this was **after restarting the local server** and without altering localstorage:

![nodes_hidden_are_wrong](https://user-images.githubusercontent.com/14145836/41182091-13d36da0-6b29-11e8-932e-19cc5b3a18ac.png)

* System retains 'notifications off' setting and
* System still detects 46 possible nodes so at least this number is consistent with previously and
* Sytem tried 42 connections which is consistent with 46 total - 4 hidden = 42

but:

* only 2 nodes are shown as hidden (despite 4 hidden nodes) and
* **these shown nodes are not the nodes I had chosen to hide** and
* Also noticed javascript `eval` errors after the initial failed OL connection was attempted (not shown in screenshot) and
* **The client finally chose to connect me to Open Ledger** (the node I am trying to avoid).

TLDR; IMHO a user should be able to choose where they want to connect, and especially during a security breach. Seems like a basic requirement and should not be overlooked.

@sschiessl-bcp do you have anything to help to understand this? I see you recently worked bounty on [10] Troubleshooting connection problems #1228 and the hidden aspect seemed to be cogent to that issue.

**Comment by sschiessl-bcp** [comment]
Connecting to nodes is handled in routerTransition.

All nodes that are considered for connection are populated here:
https://github.com/bitshares/bitshares-ui/blob/a9ee9e4c8c70c59276d19fb6b58f6bda57bda50a/app/routerTransition.js#L328

Just looking at the code it should filter out all hidden ones. May it be that clicking hide in the UI does not forward the hidden flag into the Settings properly?

**Comment by happyconcepts** [comment]
> May it be that clicking hide in the UI does not forward the hidden flag ...
I was looking at `getNodes()` line 260 ... so I thank you for your help and will have a look at the button action I guess.

I just want it to work. @startailcoon would rather fuzz my other issues than contribute help here, I guess.

**Comment by happyconcepts** [comment]
#1223

**Comment by happyconcepts** [comment]
Why would I care at this point after I fixed #1529and didn't get paid the agreed bounty? Even after the fact haters like @startailcoon are in the pull request saying that my code doesn't work, when it does.

**Comment by happyconcepts** [comment]
Since mine were the only keys stolen in the hack and no one else seems to care, I am closing this issue,
